docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+``array<str>``? If we were simply changing a scalar property to another 
+property it would be easy, because EdgeDB would prompt us for a conversion 
+expression. But a change from a link to a property is different:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+          name: str;
+          multi friends: array<str>;
+      }
+    }
+
+Doing a migration as such will just drop the ``friends`` link (along 
+with its data) and create a new ``friends`` property - without any 
+data at all.
+
+To solve this problem, we can do two migrations instead of one. First 
+we will keep the ``friends`` link, while adding a new property called 
+``friend_names``:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        name: str;
+        multi friends: User;
+        friend_names: array<str>;
+      }
+    }
+
+The CLI will simply ask us if we created a property called ``friend_names``. 
+We haven't applied the migration yet, so we might as well put the 
+data inside the same migration. A simple update will do the job!
+
+.. code-block::
+
+    CREATE MIGRATION m1hvciatdgpo3a74wagbmwhbunxbridda4qvdbrr3z2a34opks63rq
+        ONTO m1vktopcva7l6spiinh5e5nnc4dtje4ygw2fhismbmczbyaqbws7jq
+    {
+    ALTER TYPE default::User {
+        CREATE PROPERTY friend_names: array<std::str>;
+    };
+    update User set { friend_names := array_agg(.friends.name) };
+    };
+
+Now if we do a query we can confirm that the data inside ``.friends.name`` 
+when converted to an array is indeed the same as the data inside the 
+``friend_names`` property:
+
+.. code-block:: edgeql-repl
+
+    db> select User { f:= array_agg(.friends.name), friend_names };
+    {
+    default::User {
+      f: ['User 2', 'User 3'], 
+      friend_names: ['User 2', 'User 3']
+      },
+    default::User {
+      f: ['User 1', 'User 3'], 
+      friend_names: ['User 1', 'User 3']
+      },
+    default::User {
+      f: ['User 1', 'User 2'], 
+      friend_names: ['User 1', 'User 2']
+      },
+    }
+
+Or if we can't eyeball the data ourselves, we can use the ``all()`` 
+function to ensure that this is the case:
+
+.. code-block:: edgeql-repl
+
+    db> select all(array_agg(User.friends.name) = User.friend_names);
+    {true}
+
+Looks good! And now we can simply remove ``multi friends: User;`` 
+from our schema and do a final migration.
+
+Migration internals
+===================
+
+We've now reached the most optional part of the migrations tutorial, 
+but an interesting one for those curious about what goes on behind 
+the scenes during a migration.
+
+Migrations in EdgeDB before the advent of the EdgeDB Project flow 
+were still automated, but required more manual work if you didn't 
+want to accept all of the suggestions provided by the server. This 
+process is in fact still used to migrate even today; the CLI just 
+facilitates it by making it easy to respond to the generated suggestions.
+
+They took place `inside a transaction <transaction_>`_ handled by 
```

Comment:
```suggestion
Pre-project migrations took place `inside a transaction <transaction_>`_ handled by 
```
I think you've said enough here between the pronoun and its antecedent to justify naming it again.